contracts/v2/ProfitMarginProxy.sol

```diff
@@ -0,0 +1,127 @@
+        uint _minRewardsForDistribution
+    ) {
+        rewardsToken = IERC677(_rewardsToken);
+        rewardsPool = _rewardsPool;
+        ownerWallet = _ownerWallet;
+        profitMarginFeed = AggregatorV3Interface(_profitMarginFeed);
+        rewardsPoolProfitShare = _rewardsPoolProfitShare;
+        minRewardsForDistribution = _minRewardsForDistribution;
+    }
+
+    /**
+     * @dev returns the amount of available rewards
+     * @return amount of rewards
+     **/
+    function availableRewards() public view returns (uint) {
+        return rewardsToken.balanceOf(address(this));
+    }
+
+    /**
+     * @dev returns whether or not rewards should be distributed
+     * @return whether to perform upkeep
+     **/
+    function checkUpkeep(bytes calldata _checkData) external view returns (bool, bytes memory) {
+        return (availableRewards() >= minRewardsForDistribution, "0x00");
+    }
+
+    /**
+     * @dev distributes rewards if rewards >= minRewardsForDistribution
+     **/
+    function performUpkeep(bytes calldata _performData) external {
+        uint rewards = availableRewards();
+        require(rewards >= minRewardsForDistribution, "Rewards must be >= minRewardsForDistribution");
+        _distributeRewards(rewards);
+    }
+
+    /**
+     * @dev distributes rewards if there are any
+     **/
+    function distributeRewards() external {
+        uint rewards = availableRewards();
+        require(rewards > 0, "No rewards to distribute");
+        _distributeRewards(rewards);
+    }
+
+    /**
+     * @dev sets the rewardsPool address
+     * @param _rewardsPool address to set
+     **/
+    function setRewardsPool(address _rewardsPool) external onlyOwner {
+        rewardsPool = _rewardsPool;
+    }
+
+    /**
+     * @dev sets the ownerWallet address (not to be confused with the contract owner)
+     * @param _ownerWallet address to set
+     **/
+    function setOwnerWallet(address _ownerWallet) external onlyOwner {
+        ownerWallet = _ownerWallet;
+    }
+
+    /**
+     * @dev sets the address of the profit margin feed used to calculate distribution amounts
+     * @param _profitMarginFeed address to set
+     **/
+    function setProfitMarginFeed(address _profitMarginFeed) external onlyOwner {
+        profitMarginFeed = AggregatorV3Interface(_profitMarginFeed);
+    }
+
+    /**
+     * @dev sets the percentage of profits the rewardsPool receives (ownerWallet receives the rest)
+     * @param _rewardsPoolProfitShare percentage to set (2500 = 25%)
+     **/
+    function setRewardsPoolProfitShare(uint _rewardsPoolProfitShare) external onlyOwner {
+        rewardsPoolProfitShare = _rewardsPoolProfitShare;
+    }
+
+    /**
+     * @dev distributes rewards
+     * @param _amount amount of rewards to distribute
+     **/
+    function _distributeRewards(uint _amount) private {
+        (, int profitMargin, , , ) = profitMarginFeed.latestRoundData();
+
+        uint profit;
+        if (profitMargin > 0) {
```

Comment:
Require `profitMargin` to be above 0 so it's not wasting gas by continuing the method and sending 0?